client/src/ehr_components/NewPatient.tsx

```diff
@@ -0,0 +1,245 @@
+      },
+      value: formData.phoneNumber,
+      errorMessage: "Please enter a 10 digit phone number",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Address",
+      name: "address",
+      type: "Input",
+      inputType: "text",
+      placeholder: "Address",
+      onChange: updateFormData,
+      validateValue: (value: string) => {
+        return value.length > 0;
+      },
+      value: formData.address,
+      errorMessage: "Please enter a address",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Zip Code",
+      name: "zipCode",
+      type: "Input",
+      inputType: "text",
+      placeholder: "Zip Code",
+      onChange: updateFormData,
+      validateValue: (value: string) => {
+        let countDigitsRegex = /\d/g;
+        let regexMatch = value.match(countDigitsRegex);
+        return regexMatch !== null ? regexMatch.length === 5 : false;
+      },
+      value: formData.zipCode,
+      errorMessage: "Please enter a 5 digit zip code",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Job",
+      name: "job",
+      type: "Input",
+      inputType: "text",
+      placeholder: "Job",
+      onChange: updateFormData,
+      validateValue: (value: string) => {
+        return value.length > 0;
+      },
+      value: formData.job,
+      errorMessage: "Please enter a job",
+      formSubmitted: formSubmitted
+    },
+    {
+      label: "Marital Status",
+      name: "maritalStatus",
+      type: "Select",
+      value: formData.maritalStatus,
+      onChange: updateFormData,
+      placeholder: "Select Marital Status",
+      options: [
+        { value: "S", label: "Single" },
+        { value: "M", label: "Married" },
+        { value: "D", label: "Divorced" },
+        { value: "W", label: "Widow" }
+      ],
+      validateValue: (value: string) => {
+        return value.length > 0;
+      },
+      errorMessage: "Please select a marital status",
+      formSubmitted: formSubmitted
+    }
+  ];
+
+  const formIsValid = fieldsMap.reduce((previousValue, currentField) => {
+    if (currentField.validateValue) {
+      return previousValue && currentField.validateValue(currentField.value);
+    } else {
+      return previousValue;
+    }
+  }, true);
+
+  let handleClick = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    setFormSubmitted(true);
+
+    if (formIsValid) {
+      try {
+        let callResults = await Axios.post(firebaseURl, formData);
+        console.log(callResults);
+        history.push(`/main/search#success`);
+      } catch (error) {
+        toastr.error("New Patient", error.message);
+      }
+    }
+  };
+
+  return (
+    <div className="container">
+      <h2>New Patient</h2>
+      <br />
+      <div className="row  justify-content-center">
+        <form
+          className="w-75   align-content-center"
+          onSubmit={handleClick}
+          noValidate
+        >
+          {fieldsMap.map(field => {

```

Comment:
Way to go! great use of field generation

Reply:
Thank you! 🙏🏽